### PR TITLE
add /DELETE user/:id

### PR DESCRIPTION
**Pull request by Kirosana15**

### CODE REVIEW

File: src/user/test/delete.spec.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [rootMongooseTestModule(), MongooseModule.forFeature([{ name: User.name, schema: UserSchema }])],
+      providers: [UserService],
+    }).compile();
+
+    service = module.get(UserService);
+    userMock = module.get(getModelToken(User.name));
+  });
```
Comment (arkus7) [inline src/user/test/delete.spec.ts]: This could be `beforeAll()`

File: src/user/test/delete.spec.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  afterAll(async () => {
+    await closeConnections();
```
Comment (arkus7) [inline src/user/test/delete.spec.ts:29]: we should close the test module somewhere

File: src/user/test/delete.spec.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  });
+
+  it('should return deleted recipe', async () => {
```
Comment (arkus7) [inline src/user/test/delete.spec.ts]: `recipe`? it should be `user`

File: src/user/test/e2e/deleteUser.e2e.spec.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        const id = req.params._id;
+        return mockUser._id === id;
+      },
+    });
+    await app.init();
+  });
+
+  afterEach(() => {
+    app.close();
+  });
```
Comment (arkus7) [inline src/user/test/e2e/deleteUser.e2e.spec.ts]: `beforeAll`?

File: src/user/user.controller.ts [hunk 5]
```diff
@@ -17,4 +18,14 @@ export class UserController {
   getAllUsers(@Req() req: any, @Query() paginatedQueryDto?: PaginatedQueryDto): Promise<UserInfoDto[]> {
     return this.userService.getAll(paginatedQueryDto);
   }
+
+  @Delete('/:_id')
+  @UseGuards(OwnerGuard)
```
Comment (arkus7) [inline src/user/user.controller.ts:23]: What does `Owner` mean in this context? Shouldn't only admin have option to call this request?

**Review by arkus7 — APPROVED** [review]

**Review by arkus7 — CHANGES_REQUESTED** [review]
I think we should create a test app for the whole suite once, in `beforeAll` not `beforeEach`

File: src/user/user.controller.ts (hunk 5 above)
Comment (Kirosana15) [reply src/user/user.controller.ts:23]: Owner of an account. I wanted to leave the option for users to delete their own accounts. Admin can delete any account.

**Pull request merged**